src/extends/creep/actions.js

```diff
@@ -58,3 +58,62 @@ Creep.prototype.recharge = function () {
   }
   return true
 }
+
+Creep.prototype.recycle = function () {
+  let storage = this.room.storage
+  if (!storage && this.room.terminal) {
+    storage = this.room.terminal
+  }
+
+  // Empty creep of all resources, dump them in storage.
+  if (this.getCarryPercentage() > 0) {
+    if (storage) {
+      if (this.pos.isNearTo(storage)) {
+        this.transferAll(storage)
+      } else {
+        this.travelTo(storage)
+      }
+      return
+    }
+  }
+
+  // No spawn - time to suicide.
+  if (!this.room.structures[STRUCTURE_SPAWN]) {
+    this.suicide()
+    return
+  }
+
+  // Identify spawn closest to storage, to make reclaimed energy easier to store.
+  let spawn = false
+  if (storage) {
+    spawn = storage.pos.findClosestByRange(this.room.structures[STRUCTURE_SPAWN])
+  } else {
+    spawn = this.room.structures[STRUCTURE_SPAWN]
```

Comment:
I'm pretty sure this returns an array.

Reply:
Yup, fixed- I just meant to grab the first item from the array.